src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.Emitter.cs

```diff
@@ -97,9 +96,27 @@ public static void EmitRequestHandler(this Endpoint endpoint, CodeWriter codeWri
         {
             codeWriter.WriteLine("return Task.CompletedTask;");
         }
+
         codeWriter.EndBlock(); // End handler method block
     }
 
+    private static void EmitHttpResponseContentType(this EndpointResponse endpointResponse, CodeWriter codeWriter)
+    {
+        if (!endpointResponse.HasNoResponse
+            && endpointResponse.ResponseType is { } responseType
+            && (responseType.SpecialType == SpecialType.System_Object || responseType.SpecialType == SpecialType.System_String))
+        {
+            codeWriter.WriteLine("if (result is string)");
```

Comment:
Do we need to do the check at runtime if we already know the type at compile-type?

Reply:
Filters?

Reply:
This is to deal with the scenario where the resturn type on the delegate could be `object`, `Task<object>`, or `ValueTask<object>`. The actual type may be string in which case we return a `text/plain ...` content type (according to our test cases, and RDF).

Reply:
... the implication of the above is that what actual type that is returned could be changed at runtime so we need a runtime check.